src/species/rootworm.ts

```diff
@@ -0,0 +1,221 @@
+      name: 'wings',
+      default: 0
+    }), new Trait({
+      name: 'energy',
+      default: 1000
+    }), new Trait({
+      name: 'wandering threshold',
+      default: 5
+    }), new Trait({
+      name: 'hunger bonus',
+      default: 15
+    })
+  ];
+
+class WormAnimal extends BasicAnimal {
+
+  protected step() {
+    super.step();
+  }
+  protected eat() {
+    const nearest = super._nearestPrey();
+    // Until we have other plants, assume nearest prey is corn
+    if (nearest) {
+      const eatingDist = super.get('eating distance');
+      if (nearest.distanceSq < Math.pow(eatingDist, 2)) {
+        // we're close enough, eat corn
+        this._eatCorn(nearest.agent);
+      }
+      else {
+        // move to corn
+        this.chase(nearest);
+      }
+    }
+    else {
+      super.wander(super.get('speed') * 0.75);
+    }
+  }
+
+  private _eatCorn(cornAgent: IAgent) {
+    const consumptionRate = super.get('resource consumption rate');
+    const currEnergy = super.get('energy');
+    super.set('energy', currEnergy + consumptionRate);
+
+    const cornHealth = cornAgent.get('health');
+    const newHealth = cornHealth - consumptionRate;
+    if (newHealth > 10) {
+      cornAgent.set('health', cornHealth - consumptionRate);
+    }
+    else {
+      cornAgent.die();
+    }
+  }
+  protected chase(nearestAgent: any) {
+    super.chase(nearestAgent);
+  }
+  protected move(speed: any) {
+    // energy currently used as a health-meter for bugs
+    const currEnergy = super.get('energy');
+
+    if (currEnergy === 0) {
+      super.die();
+    }
+    if (super.get('current behavior') !== 'eating') {
+      if (super.get('age') < maturity) {
+        // larva can't move fast
+        speed = speed > super.get('larva max speed') ? super.get('larva max speed') : speed;
+      } else {
+        // adults that fly will use up energy more quickly
+        super.set('energy', currEnergy - (speed * 0.5));
+      }
+      super.move(speed);
+    } else {
+      // we're eating, move slowly
+      super.move(speed * 0.01);
+    }
+  }
+}
+
+export const worm: ISpecies = new Species({
+  speciesName: "Worm",
+  agentClass: WormAnimal,
+  defs: {
+    CHANCE_OF_MUTATION: 0,
+    MATURITY_AGE: maturity
+  },
+  traits: wormTraits,
+  imageRules: [
+    {
+      name: 'worm',
+      rules: [
+        {
+          image: {
+            path: require('../images/rootworm-larva1.png'),
+            scale: 0.4,
+            anchor: {
+              x: 0.4,
+              y: 1
+            }
+          },
+          useIf(agent: IAgent) {
+            return agent.get('age') < 10;
+          }
+        },
+        {
+          image: {
+            path: require('../images/rootworm-larva1.png'),
+            scale: 0.5,
+            anchor: {
+              x: 0.5,
+              y: 1
+            }
+          },
+          useIf(agent: IAgent) {
+            return agent.get('age') >= 10 && agent.get('age') < (maturity * 0.25);
```

Comment:
Seems like it might be a good idea to define constants for these thresholds so that if we ever need to tweak them, they only have to be tweaked in one place rather than trying to make sure that they're tweaked consistently in multiple places.